content_scripts/inject_script.coffee

```diff
@@ -0,0 +1,75 @@
+# We can't inject inline scripts into our chrome-extension pages (eg. options), so don't try.
+return if window.location.protocol == "chrome-extension:"
+
+injectScripts = [
+  ( -> # Hook addEventListener to tell link hints when a click listener is added.
+    _addEventListener = Element::addEventListener
+
+    elementsToRegister = []
+    registrationElement = null
+
+    EventTarget::addEventListener = (type, listener, useCapture) ->
+      eventTarget = if this in [document, window] then document.documentElement else this
+      if type == "click" and eventTarget instanceof Element
+        setTimeout (-> registerElementWithContentScripts eventTarget, "onclick"), 0
+      _addEventListener.apply this, arguments
+
+    onLoaded = ->
+      # Create an element detatched from the DOM, register it with the content scripts.
+      registrationElement = document.createElementNS "http://www.w3.org/1999/xhtml", "div"
+      registrationEvent = new CustomEvent "VimiumRegistrationElementEvent"
+      document.documentElement.appendChild registrationElement
+      registrationElement.dispatchEvent registrationEvent
+      document.documentElement.removeChild registrationElement
+
+      elementsToRegister.map (args) ->
```

Comment:
Warning: `Array::map` may be rewritten by other websites and not work. Example: In http://jsfiddle.net/p7w9yvsn/2/, the top frame rewrites `Function::bind`, which causes my custom Chrome extension fail again and again.